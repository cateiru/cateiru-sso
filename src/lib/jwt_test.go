package lib_test

import (
	"testing"

	"github.com/cateiru/cateiru-sso/src/lib"
	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/stretchr/testify/require"
)

const JWT_PUBLIC_KEY_PATH = "jwt/jwt.pkcs8"

func TestJsonWebKeys(t *testing.T) {
	resp, err := lib.JsonWebKeys(JWT_PUBLIC_KEY_PATH, "RS256", "sig", "test")
	require.NoError(t, err)

	snaps.MatchSnapshot(t, resp)
}
