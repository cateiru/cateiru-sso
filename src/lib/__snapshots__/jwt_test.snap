
[TestJsonWebKeys - 1]
&&jwk.rsaPublicKey{
    algorithm:              &"RS256",
    e:                      {0x1, 0x0, 0x1},
    keyID:                  &"test",
    keyOps:                 (*jwk.KeyOperationList)(nil),
    keyUsage:               &"sig",
    n:                      {0xa2, 0x34, 0x97, 0x36, 0xb6, 0x20, 0xc1, 0xf, 0xc4, 0xe6, 0x54, 0x13, 0x10, 0xeb, 0x9e, 0x77, 0xb7, 0xb6, 0xea, 0xf7, 0x48, 0xa2, 0x2f, 0xe5, 0xba, 0xb4, 0x73, 0x43, 0x26, 0x61, 0x74, 0x48, 0xb0, 0x5b, 0xf8, 0xb1, 0x9b, 0x85, 0x5b, 0x5e, 0xba, 0xc0, 0x35, 0x1b, 0xee, 0xc8, 0x4f, 0x4a, 0x81, 0xa8, 0x45, 0xa1, 0xd2, 0x8e, 0xce, 0xd0, 0x15, 0xf3, 0xb9, 0xd7, 0xe0, 0xf5, 0x6b, 0x99, 0x6e, 0x23, 0x1c, 0xa5, 0x13, 0xac, 0xea, 0xea, 0x60, 0xf1, 0xb9, 0xc4, 0x13, 0x50, 0xa5, 0xb6, 0x9c, 0xbc, 0xe5, 0x22, 0x88, 0x74, 0xc8, 0xf, 0x6a, 0xfc, 0x7b, 0xf0, 0xd9, 0xba, 0xe1, 0xd5, 0x55, 0xfe, 0xb5, 0x1a, 0xf8, 0xe5, 0xf9, 0xfe, 0x1a, 0x40, 0xa7, 0x18, 0xf8, 0x2c, 0x99, 0x25, 0x9f, 0xda, 0x63, 0xa3, 0x11, 0x6d, 0x5e, 0xe4, 0x90, 0x6a, 0xc0, 0xf1, 0x28, 0x99, 0xd1, 0x86, 0x20, 0x3c, 0xe9, 0xc4, 0x14, 0xa4, 0x22, 0x9d, 0x18, 0x4a, 0xf8, 0x9e, 0xad, 0x77, 0xab, 0x9f, 0x87, 0x5d, 0x9e, 0x18, 0xb9, 0x73, 0x74, 0xac, 0xd4, 0x65, 0xeb, 0x75, 0xcf, 0x4e, 0x59, 0x81, 0x4a, 0x75, 0x24, 0x9, 0xf5, 0xb0, 0x8c, 0x7f, 0x1f, 0x6c, 0x69, 0x9e, 0xa5, 0x94, 0x8f, 0xc9, 0xd4, 0xfe, 0x65, 0xdd, 0x31, 0x25, 0xe4, 0xb7, 0xe5, 0x7, 0xe0, 0xa2, 0xe5, 0x6, 0x1a, 0xa0, 0x6f, 0x7, 0xcb, 0xd, 0x6c, 0x51, 0xfa, 0x8d, 0x41, 0xb9, 0xd, 0x1c, 0x4, 0x3c, 0x44, 0xbf, 0xd8, 0xb, 0x24, 0x8c, 0x81, 0x6f, 0x59, 0x8b, 0xe9, 0x83, 0xb3, 0x87, 0x10, 0x23, 0x12, 0x8b, 0x4f, 0xbd, 0x4a, 0x37, 0xc1, 0x0, 0xec, 0x12, 0xbf, 0x92, 0x7a, 0xb0, 0x41, 0xd9, 0x57, 0x14, 0x39, 0x7c, 0xc5, 0x66, 0x8e, 0x28, 0xf, 0xec, 0xd1, 0xdc, 0x8e, 0x3d, 0xa3, 0xcf, 0x37, 0xf3, 0x6b, 0xf5, 0xe0, 0x89, 0xd1, 0xfd, 0x1a, 0xb7, 0xfd, 0xb1, 0x3c, 0x42, 0x11, 0x3, 0xc1, 0x5b, 0x0, 0xa1, 0x55, 0x35, 0x8f, 0xb0, 0x64, 0x7c, 0x16, 0xc2, 0x88, 0x69, 0xb7, 0x51, 0xf4, 0x25, 0xd3, 0x74, 0xdc, 0x6f, 0x97, 0x83, 0xcd, 0x21, 0xd, 0xd1, 0x15, 0x1a, 0xa0, 0xf4, 0x90, 0x4b, 0xf7, 0xe3, 0xdf, 0x31, 0x1c, 0xbf, 0x56, 0x8a, 0x63, 0x44, 0xf7, 0xd5, 0xfb, 0xd4, 0x29, 0xe, 0x68, 0xf1, 0xa3, 0x12, 0xdd, 0xe8, 0x8c, 0xd1, 0x74, 0x17, 0x94, 0xae, 0x95, 0xf9, 0x92, 0xe0, 0x1f, 0xca, 0x3e, 0x76, 0xc1, 0x40, 0x4b, 0x63, 0x8a, 0xf5, 0x11, 0x42, 0x53, 0xe6, 0xa7, 0x21, 0xe4, 0xdf, 0x20, 0xf5, 0xc4, 0x64, 0x70, 0xda, 0x99, 0x58, 0xbc, 0xc9, 0x18, 0xfe, 0x56, 0xd0, 0x8e, 0x28, 0x98, 0xf8, 0x69, 0x39, 0x93, 0x40, 0x0, 0x58, 0x2b, 0x45, 0x10, 0x3d, 0x89, 0xa9},
    x509CertChain:          (*cert.Chain)(nil),
    x509CertThumbprint:     (*string)(nil),
    x509CertThumbprintS256: (*string)(nil),
    x509URL:                (*string)(nil),
    privateParams:          {
    },
    mu: &sync.RWMutex{},
    dc: nil,
}
---
